packages/features/tasker/tasks/scanWorkflowBody.ts

```diff
@@ -55,38 +40,66 @@ export async function scanWorkflowBody(payload: string) {
     },
   });
 
-  for (const workflowStep of workflowSteps) {
-    if (!workflowStep.reminderBody) {
-      await prisma.workflowStep.update({
-        where: {
-          id: workflowStep.id,
-        },
-        data: {
-          verifiedAt: new Date(),
-        },
+  if (process.env.IFFY_API_KEY) {
+    for (const workflowStep of workflowSteps) {
+      if (!workflowStep.reminderBody) {
+        await prisma.workflowStep.update({
+          where: {
+            id: workflowStep.id,
+          },
+          data: {
+            verifiedAt: new Date(),
+          },
+        });
+        continue;
+      }
+
+      const timeFormat = getTimeFormatStringFromUserTimeFormat(workflowStep.workflow.user?.timeFormat);
+
+      // Determine if body is a template
+      const defaultTemplate = getTemplateBodyForAction({
+        action: workflowStep.action,
+        locale: workflowStep.workflow.user?.locale ?? "en",
+        t: await getTranslation(workflowStep.workflow.user?.locale ?? "en", "common"),
+        template: workflowStep.template,
+        timeFormat,
       });
-      continue;
-    }
 
-    const timeFormat = getTimeFormatStringFromUserTimeFormat(workflowStep.workflow.user?.timeFormat);
+      if (!defaultTemplate) {
+        log.error(`Template not found for action ${workflowStep.action}, template ${workflowStep.template}`);
+        continue;
+      }
 
-    // Determine if body is a template
-    const defaultTemplate = getTemplateBodyForAction({
-      action: workflowStep.action,
-      locale: workflowStep.workflow.user?.locale ?? "en",
-      t: await getTranslation(workflowStep.workflow.user?.locale ?? "en", "common"),
-      template: workflowStep.template,
-      timeFormat,
-    });
+      if (
+        compareReminderBodyToTemplate({ reminderBody: workflowStep.reminderBody, template: defaultTemplate })
+      ) {
+        await prisma.workflowStep.update({
+          where: {
+            id: workflowStep.id,
+          },
+          data: {
+            verifiedAt: new Date(),
+          },
+        });
+        continue;
+      }
 
-    if (!defaultTemplate) {
-      log.error(`Template not found for action ${workflowStep.action}, template ${workflowStep.template}`);
-      continue;
-    }
+      const isSpam = await iffyScanBody(workflowStep.reminderBody, workflowStep.id);
+
+      if (isSpam) {
+        if (!workflowStep.workflow.user?.whitelistWorkflows) {
+          // We won't delete the workflow step incase it is flagged as a false positive
+          log.warn(`Workflow step ${workflowStep.id} is spam with body ${workflowStep.reminderBody}`);
+          await lockUser("userId", userId.toString(), LockReason.SPAM_WORKFLOW_BODY);
+
+          // Return early if spam is detected
+          return;
+        }
+        log.warn(
+          `For whitelisted user, workflow step ${workflowStep.id} is spam with body ${workflowStep.reminderBody}`
+        );
```

Comment:
Q: we don't need to do any action here?

Reply:
No, for whitelisted users, we log that warning, and below we are setting `verifiedAt`